assets/src/data/eventespresso/lists/reducers.js

```diff
@@ -6,19 +6,46 @@ import { DEFAULT_LISTS_STATE } from '../../model';
 /**
  * External dependencies
  */
-import { isEqual } from 'lodash';
+import isShallowEqual from '@wordpress/is-shallow-equal';
 
 /**
  * Returns whether the state matches the provided items.
- * @param { Object } state
- * @param { string } modelName
- * @param { string } queryString
- * @param { Array } items
- * @return { boolean } If the items are in state and they match, then true.
+ * @param {Object} state
+ * @param {string} identifier
+ * @param {string} queryString
+ * @param {Array} items
+ * @return {boolean} If the items are in state and they match, then true.
  */
-const stateMatchesItems = ( state, modelName, queryString, items = [] ) => (
-	state[ modelName ].hasOwnProperty( queryString ) &&
-	isEqual( state[ modelName ][ queryString ], items )
+const stateMatchesItems = ( state, identifier, queryString, items = [] ) => (
+	state[ identifier ] &&
+	state[ identifier ][ queryString ] &&
+	isShallowEqual( state[ identifier ][ queryString ], items )
+);
+
+/**
+ * Returns whether there is a match for incoming entities against what is
+ * currently in state.  The match is performed against the keys (ids) for the
+ * entities
+ *
+ * @param {Object} state
+ * @param {string} modelName
+ * @param {string} queryString
+ * @param { Map<number|string,Object>} entities
+ * @return {boolean} If the incoming entity keys match the entity keys currently
+ * in the state for the given queryString and modelName then this returns true.
+ */
+const stateMatchesEntities = (
```

Comment:
> Wouldn't using a simple Object result in duplicates,

I'm not sure I understand what you mean here.  What happens is something like this:

```js
const someObject = {};
someObject['20'] = 'foo';
someObject['10'] = 'bar';
// { 10: foo, 20: bar }
console.log( someObject );
```

Same thing with non-numeric keys, there's no guarantee that the order you add things to the object is preserved.  Hence the need for a `Map` because it DOES preserve order (plus keys can be pretty much anything and type is preserved).